octobot_trading/exchanges/types/rest_exchange.py

```diff
@@ -163,6 +163,45 @@ async def _edit_order(self, order_id: str, order_type: enums.TraderOrderType, sy
                                                quantity, price, stop_price, side,
                                                current_price, params)
 
+    async def edit_order_by_replacing(
+        self,
+        order_id: str,
+        order_type: enums.TraderOrderType,
+        symbol: str,
+        quantity: decimal.Decimal,
+        price: decimal.Decimal,
+        stop_price: decimal.Decimal = None,
+        side: enums.TradeOrderSide = None,
+    ):
+        # Can be used if api doesnt have an endpoint to edit orders
+        # see binance USD m as an example
+        order_to_cancle = (
+            self.exchange_manager.exchange_personal_data.orders_manager.get_order(
+                order_id
+            )
+        )
+        await self.cancel_order(order_id=order_id, symbol=symbol)
+        await order_to_cancel.state.wait_for_terminate(
+            constants.INDIVIDUAL_ORDER_SYNC_TIMEOUT
+            )
+        replaced_order = await self.create_order(
+            order_type,
+            symbol,
+            quantity,
+            price=stop_price or price,
+            stop_price=stop_price,
+            side=side,
+            current_price=order_to_cancle.created_last_price,
+            reduce_only=order_to_cancle.reduce_only,
+        )
+        if replaced_order:
+            return replaced_order
+        else:
+            raise RuntimeError(
```

Comment:
I think we should prefer something like a 
```suggestion
            raise errors.FailedRequest(
```